GSoC contributor blog for Unreleased Production changes probe

Jagrutiti opened the pull request:
Submitting blog for "Unreleased Production changes probe" which is live into production as a part of GSoC 2023 activity.

Jagrutiti commented:
@alecharp, @dheerajodha, @aneveux , @krisstern requesting to review the bog when possible

dheerajodha commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,60 @@
+
+== Conclusion
+
+I once again realised that open-source world values quality. They are fine with fewer quality probes than many not so useful probes.
```
```suggestion
I once again realised that the open-source world values quality. They are fine with fewer "quality" probes than many "not-so-useful" probes.
```
optional (feel free to ignore): but "fewer quality probes" kind of reads like "they are okay with probes with fewer quality".

alecharp replied:
With a new line after the end of the sentence ;)

dheerajodha commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,60 @@
+---
+
+== Introduction
+Unreleased production changes probe looks for unreleased commits in the production files.
```
```suggestion
The "unreleased production changes" probe looks for unreleased commits in the production files.
```
reads a bit different without the quotes.

dheerajodha commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,60 @@
+
+* The plugin may have undelivered bug fixes.
+* The plugin may have stale features.
+* The maintainer of the plugin maybe inactive.
```
```suggestion
* The maintainer of the plugin may be inactive.
```

dheerajodha commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,60 @@
+In the end, my mentors suggested a pair programming session. We rewrote the probe together. I am thankful to their kindness and patience.
+
+== Outcome
+In the case when unreleased production files are found, the probe returns a list with their names and a failure message is returned. In the case when no unreleased production file is found, the probe returns a success message.
```
```suggestion
In the case when unreleased production files are found, the probe returns a list with their names, and an overall failure message is returned. In the case when no unreleased production file is found, the probe returns a success message.
```
optional

alecharp replied:
Not quite
```suggestion
In the case when unreleased production files are found, the probe returns failure status and the list of files name.
In the case when no unreleased production file is found, the probe returns a success message.
```

alecharp commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,60 @@
+== Introduction
+Unreleased production changes probe looks for unreleased commits in the production files.
+
+If an unreleased file is found in the production path, the probe returns a failure message. If no unreleased file is found, it returns a success message.
```
You can ignore this.

```suggestion
If an unreleased file is found in the production path, the probe returns a failure message.
If no unreleased file is found, it returns a success message.
```

alecharp commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,60 @@
+This is my first probe that went live as a GSoC contributor.
+
+== Importance of the probe
+Finding an unreleased production file can mean one or more of the following:
+
+* The plugin may have undelivered bug fixes.
+* The plugin may have stale features.
+* The maintainer of the plugin maybe inactive.
```
I feel it would be better to tell the story here rather than listing items.

Jagrutiti replied:
Thank you for suggesting the story thing. It makes the blog more interesting to read.

alecharp commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,60 @@
+* The plugin may have stale features.
+* The maintainer of the plugin maybe inactive.
+
+== Probe details
+The probe aims to identify unreleased production commits in `src/main` directory and `pom.xml` file.
+
```
This could be removed and included in the `Importance of the probe` section.

alecharp commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,60 @@
+The probe aims to identify unreleased production commits in `src/main` directory and `pom.xml` file.
+
+== Challenges
+Implementing https://www.eclipse.org/jgit/[JGit], a Java library that implements Git version control system was one of the biggest challenges I faced when working on this probe. Also, I was not familiar with Git hierarchy. I struggled to find the parent of a child commit using JGit.
```
```suggestion
Implementing https://www.eclipse.org/jgit/[JGit], a Java library that implements Git version control system was one of the biggest challenges I faced when working on this probe.
Also, I was not familiar with Git, how the commits are linked together.
Because of that, I struggled to find the parent of a child commit using JGit.
```

alecharp commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,60 @@
+== Probe details
+The probe aims to identify unreleased production commits in `src/main` directory and `pom.xml` file.
+
+== Challenges
+Implementing https://www.eclipse.org/jgit/[JGit], a Java library that implements Git version control system was one of the biggest challenges I faced when working on this probe. Also, I was not familiar with Git hierarchy. I struggled to find the parent of a child commit using JGit.
+
+In the end, my mentors suggested a pair programming session. We rewrote the probe together. I am thankful to their kindness and patience.
```
Ignore if you want

```suggestion
In the end, my mentors suggested a pair programming session in order to review thinking process involved in implementing such probe.
I am thankful to their kindness and patience.
```

kmartens27 replied:
```suggestion
== Challenges
Implementing link:https://www.eclipse.org/jgit/[JGit], a Java library that implements a Git version control system, was one of the biggest challenges I faced when working on this probe.
Since I was not familiar with Git hierarchy, I struggled to find the parent of a child commit using JGit.

In the end, my mentors suggested a pair programming session to review the thinking process involved in implementing such a probe.
I am thankful for their kindness and patience.
```
suggestion to update syntax so that the sentence reads smoothly

alecharp commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,60 @@
+== Outcome
+In the case when unreleased production files are found, the probe returns a list with their names and a failure message is returned. In the case when no unreleased production file is found, the probe returns a success message.
+
+== Learning
+* Learned to figure out unit test cases and implemented them.
+* Implemented JGit Java library.
+* Understood Git commit hierarchy.
```
Same as before, telling the story rather than listing the items could be better.

alecharp commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,60 @@
+* Understood Git commit hierarchy.
+
+== Conclusion
+
```
To be coherent with the rest of the document

```suggestion
```

alecharp commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,60 @@
+I once again realised that open-source world values quality. They are fine with fewer quality probes than many not so useful probes.
+
+Starting from ground zero is tough. Be it implementing a new Java library or starting new work in some other field. With correct mentorship and guidance you can rise up faster.
+
+
+
+
+
```
```suggestion
```

alecharp commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,60 @@
+
+== Conclusion
+
+I once again realised that open-source world values quality. They are fine with fewer quality probes than many not so useful probes.
+
+Starting from ground zero is tough. Be it implementing a new Java library or starting new work in some other field. With correct mentorship and guidance you can rise up faster.
```
```suggestion
Starting from ground zero is tough.
Be it implementing a new Java library or starting new work in some other field.
With correct mentorship and guidance you can rise up faster.
```

kmartens27 replied:
```suggestion
I once again realised the open-source world values quality.
The community is fine with fewer "quality" probes, as oposed to many "not-so-useful" probes.

Starting from ground zero is tough, whether implementing a new Java library or starting new work in another field.
However, with correct mentorship and guidance, you can rise up faster.
```
suggestion to make the sentences read a bit better/update syntax

alecharp reviewed (changes requested):
I think you can improve the story on this blog entry to better convey some messages.

kmartens27 commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,60 @@
+In the end, my mentors suggested a pair programming session. We rewrote the probe together. I am thankful to their kindness and patience.
+
+== Outcome
+In the case when unreleased production files are found, the probe returns a list with their names and a failure message is returned. In the case when no unreleased production file is found, the probe returns a success message.
```
```suggestion
When unreleased production files are found, the probe returns a failure status and the list of filenames.
When no unreleased production file is found, the probe returns a success message.
```
suggestion to remove the "In the case" and start sentences sooner.

kmartens27 reviewed (commented):
Thanks very mcuh @Jagrutiti, I agree with a few of the suggestions that @alecharp made, and added a few of my own to help. If there's anything you do not agree with or have questions on, let me know and I'm happy to help further!

zbynek commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,60 @@
+- gsoc
+---
+
+== Introduction
```
Since this is a blogpost, it would be useful to give some context: mention that you're building this as part of a GSoC project, add a link to your project description.

zbynek commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,60 @@
+The probe aims to identify unreleased production commits in `src/main` directory and `pom.xml` file.
+
+== Challenges
+Implementing https://www.eclipse.org/jgit/[JGit], a Java library that implements Git version control system was one of the biggest challenges I faced when working on this probe.
```
To me "implementing JGit" sounds like you've done some changes to JGit. Maybe "learning to use JGit" would describe the challenge better?

Jagrutiti commented:
> Thanks very mcuh @Jagrutiti, I agree with a few of the suggestions that @alecharp made, and added a few of my own to help. If there's anything you do not agree with or have questions on, let me know and I'm happy to help further!

I have added all the suggested fixes. Let me know what do you think.

krisstern commented:
@alecharp Please review this PR again and approve so that it can be merged

kmartens27 commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,81 @@
+
+== Introduction
+
+A bug fix, security fix, new features are of no use unless they are released.
```
```suggestion
A bug fix, security fix, or new features are of no use unless they are released.
```
Suggestion to add a word for readability

kmartens27 commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,81 @@
+
+A bug fix, security fix, new features are of no use unless they are released.
+
+A production release not only provides users with updated plugin but also give the developers a chance to get feedback on their work.
```
```suggestion
A production release not only provides users with an updated plugin, but also gives the developers a chance to get feedback on their work.
```
suggestions for syntax & punctuation

kmartens27 commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,81 @@
+If an unreleased file is found in the production path, the probe returns a failure message.
+If no unreleased file is found, it returns a success message.
+
+NOTE: for now, we are focusing on Maven production files (`pom.xml` and `src/main` folder). 
```
```suggestion
NOTE: For now, we are focusing on Maven production files (`pom.xml` and `src/main` folder). 
```
suggestion for capitalization

kmartens27 commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,81 @@
+
+* There might be unreleased security fixes that should be released as soon as possible.
+* There might be unreleased features requested by active users to improve the usefulness of the plugin.
+* One of the reasons for unreleased production changes could be an inactive or a super busy maintainer. This means the plugin needs a new maintainer, and it should be up for adoption.
```
```suggestion
* The plugin maintainer may be inactive or too busy to further develop the plugin.
This can indicate the plugin needs a new maintainer, and it should be up for adoption.
```
Suggestion to rephrase to make this more inline with the two previous statements. If this is not the intended message, disregard!

kmartens27 commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,81 @@
+When no unreleased production file is found, the probe returns a success message.
+
+== Learning
+I hope there comes a day when test cases becomes my friends. But I don't see this happening anytime soon.
```
```suggestion
I hope there comes a day when test cases becomes my friends, but I don't see this happening anytime soon.
```
suggestion for punctuation/sentence structure

kmartens27 commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,81 @@
+== Learning
+I hope there comes a day when test cases becomes my friends. But I don't see this happening anytime soon.
+
+Mentors suggested I should start with writing the test cases to reproduce the probe requirement. And later add code to pass each of the test cases.
```
```suggestion
Mentors suggested I should start with writing the test cases to reproduce the probe requirement, and later add code to pass each of the test cases.
```
suggestion for sentence structure

kmartens27 commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,81 @@
+
+Mentors suggested I should start with writing the test cases to reproduce the probe requirement. And later add code to pass each of the test cases.
+
+This gave me understanding of what the probe should actually do. And I learned to identify the test case scenarios.
```
```suggestion
This gave me an understanding of what the probe should actually do, and I learned to identify the test case scenarios.
```
suggestion for sentence structure/syntax

kmartens27 commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,81 @@
+
+Reading commits wasn't possible without JGit Java library.
+
+To implement JGit library and understand the features it provides, you should have strong understanding of Git concepts. I lacked depth in Git concepts that made me struggle with JGit.
```
```suggestion
To implement JGit library and understand the features it provides, you should have strong understanding of Git concepts.
I lacked depth in Git concepts that made me struggle with JGit.
```
suggestion for document formatting

kmartens27 commented on content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,81 @@
+However, with correct mentorship and guidance, you can rise up faster.
+
+== Links
+For more information or to find answers to any questions you might have, head to the official GSoC 2023 project link:https://www.jenkins.io/projects/gsoc/2023/projects/add-probes-to-plugin-health-score/[Adding Probes to  "Plugin Health Score"] description page.
```
```suggestion
For more information or to find answers to any questions you might have, head to the official GSoC 2023 project link:/projects/gsoc/2023/projects/add-probes-to-plugin-health-score/[Adding Probes to  "Plugin Health Score"] description page.
```
suggestion to use a direct link, as this would be on the existing site and does not need the full URL.

kmartens27 reviewed (approved):
This is great @Jagrutiti, thanks again for writing this up!

I've provided some suggestions but are mostly sentence structure/formatting, so nothing major. Once you've had a chance to review and if you have any additional questions, I'm happy to help in any way.

Outside of that, would you be able to change the corresponding folders & filename so that it uses content/blog/2023/07/13 and the filename is "2023-07-13-unreleased-production-changes-probe.adoc"? That will ensure that it is located in the correct location.

kmartens27 commented on content/blog/2023/07/16/2023-07-16-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,83 @@
+
+Finding an unreleased production file can mean one or more of the following:
+
+* There might be unreleased security fixes that should be released as soon as possible.
+* There might be unreleased features requested by active users to improve the usefulness of the plugin.
+* The plugin maintainer may be inactive or too busy to further develop the plugin.
+This can indicate the plugin needs a new maintainer, and it should be up for adoption.
```
```suggestion
* There are unreleased security fixes that should be released as soon as possible.
* There are unreleased features requested that would improve the usefulness of the plugin.
* The plugin maintainer may be inactive or too busy to further develop the plugin.
This could indicate the plugin needs a new maintainer and should be up for adoption.
```
suggestion for sentence wording

kmartens27 reviewed (commented):
Hi @Jagrutiti, I had a couple final notes/suggestions to make the text a bit more clear, but didn't want to change anything myself.

If you're okay with the changes/want to implement them, everything should be all set after that and we can publish the blog post!

krisstern commented on content/blog/2023/07/16/2023-07-16-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,83 @@
+When no unreleased production file is found, the probe returns a success message.
+
+== Learning
+I hope there comes a day when test cases becomes my friends, but I don't see this happening anytime soon.
```
```suggestion
I hope there comes a day when test cases become my friends, but I don't see this happening anytime soon.
```

krisstern commented on content/blog/2023/07/16/2023-07-16-unreleased-production-changes-probe.adoc:
```diff
@@ -0,0 +1,83 @@
+== Learning
+I hope there comes a day when test cases becomes my friends, but I don't see this happening anytime soon.
+
+Mentors suggested I should start with writing the test cases to reproduce the probe requirement, and later add code to pass each of the test cases.
```
```suggestion
Mentors suggested I should start with writing the test cases to reproduce the probe requirement(s), and later add code to pass each of the test cases.
```
